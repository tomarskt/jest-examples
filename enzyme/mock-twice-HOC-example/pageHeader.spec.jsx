import React from 'react';
import renderer from 'react-test-renderer';
import PageHeader from './pageHeader';

jest.mock('react-router', () => ({
  ...require.requireActual('react-router'), // leave the rest untouched... only mock what impact us
  withRouter: Component => props => (
    <Component location={{ pathname: '/' }} {...props} />
  ),
}));

describe('PageHeader Test', () => {
  it('should match snapshot for simple PageHeader', () => {
    const wrapper = renderer
      .create(
            <PageHeader />
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
