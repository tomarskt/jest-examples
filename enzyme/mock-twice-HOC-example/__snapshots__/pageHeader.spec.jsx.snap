// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Test should match snapshot for simple PageHeader 1`] = `
<div>
  <h1>
    hamburger menu
  </h1>
</div>
`;
